diag_manager/fms_diag_yaml.F90

```diff
@@ -441,6 +443,8 @@ subroutine diag_yaml_object_init(diag_subset_output)
       !> Save the variable name and the module name in the variable_list
       variable_list%var_name(var_count) = trim(diag_yaml%diag_fields(var_count)%var_varname)//&
                                         ":"//trim(diag_yaml%diag_fields(var_count)%var_module)//c_null_char
+      !! The diag_table is not case sensitive (so we are saving it as lowercase)
+      variable_list%var_name(var_count) = lowercase(variable_list%var_name(var_count))
```

Comment:
They **can** have capital letters. 

This is for when we are searching for the variable name in the diag_table.yaml. The old diag manager was not case sensitive. So in the `register_diag_field` call the variable name could be "Temperature", and in the diag_table.yaml "TEMPERATURE" or whatever. Without this the code won't be able to find the variable because `"Temperature" != "TEMPERATURE"`

The variable_list is the sorted list of variables, in the diag_yaml it saved at whatever is in the diag_table.yaml